modules/kafka/src/main/scala/com/dimafeng/testcontainers/KafkaContainer.scala

```diff
@@ -13,25 +16,45 @@ case class KafkaContainer(dockerImageName: DockerImageName = DockerImageName.par
 
 object KafkaContainer {
 
-  val defaultImage = Option(System.getProperty("os.arch")) match {
-    /* 
+  private lazy val runningOnArm64: Boolean =
+    // This shouldn't fail (`null` if it doesn't exist), however just in case...
+    // `os.arch` is `x86_64` if we are on an x86 machine _or_ the JVM is running under Rosetta 2 (emulation).
+    // `aarch64` if we are on ARM64 _and_ not running under Rosetta 2 (emulation).
+    Try(System.getProperty("os.arch")).toOption.exists {
+      // We know we definitely are on an ARM64 processor
+      case "aarch64" => true
+      // We may be on ARM64 if the JVM is being emulated, find if it is running under Rosetta 2
+      case _ => runningUnderRosetta2
+    }
+
+  // `sysctl -n sysctl.proc_translated` is defined on Apple ARM64 devices.
+  // It returns `0` when running natively and `1` when running the current process under Rosetta 2.
+  private lazy val runningUnderRosetta2: Boolean =
+    // ProcessLogger redirects any error output away from console to a noop receiver (effectively `/dev/null`)
+    Try("sysctl -n sysctl.proc_translated".!!(ProcessLogger(_ => ()))).toOption.exists(_.trim == "1")
```

Comment:
Two quick questions:
1. I'm curious how this is handled in the Java version of this container? Can we reuse it from there?
2. Will it work on windows? Per my understanding it'll go to "runningUnderRosetta2" execution path

Reply:
1. Ah, good question, I'll take a look and report back :+1:
2. I don't have a windows machine to hand but I am 99% sure it will return `false`. The `Try` block should fail (command will fail), so the `Option` is empty and `.exists` always returns `false` in an empty option

Reply:
Ah, if you mean that it will hit the `runningUnderRosetta2` path than you are correct, sorry. I am just assuming that the code block will subsequently fail for any number of reasons, so we'll return `false`.

Reply:
got it. I makes sense. Let's see how the java lib handles it and then leave it as it if they don't have a better solution.

Reply:
From what I can see `testcontainers-java` does nothing special at all: https://github.com/testcontainers/testcontainers-java/blob/dace7e4d750552bb3d57036b579318045e203931/modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

Reply:
Got it! Thanks. Just one last thought - would it make sense to contribute this change to the java library so it’s consistently available in the java and Scala versions?